### ISSUE TITLE
No alignment

### ISSUE THREAD

**Issue by portbo** [paoloshasta/shasta#36]
Hi,
I have a ~130kb N50 single flow cell out put from a UL ONT chemistry 10

Post inital qc Shasta identifies the following: The run finishes without errors but there is no alignment and the fasta is empty:

Command line:
./shasta-Linux-0.12.0 --input /oxford_nanopore_UL/E13.pass.fq --config Nanopore-ncm23-May2024 --Align.alignMethod 3 --assemblyDirectory /oxford_nanopore_UL/output2
For options in use for this assembly, see shasta.conf in the assembly directory.This run uses options "--memoryBacking 4K --memoryMode anonymous".
This could result in longer run time.
For faster assembly, use "--memoryBacking 2M --memoryMode filesystem"
(root privilege via sudo required).
Therefore the results of this run should not be used
for the purpose of benchmarking assembly time.
However the memory options don't affect assembly results in any way.
This assembly will use 48 threads.
Turning off --Reads.noCache for/oxford_nanopore_UL/E13.pass.fq
Discarded read statistics for file /oxford_nanopore_UL/E13.pass.fq:
    Discarded 0 reads containing invalid bases for a total 0 valid bases.       
    Discarded 206363 reads shorter than 10000 bases for a total 748307569 bases.    Discarded 0 reads containing repeat counts 256 or more for a total 0 bases. 
Discarded read statistics for all input files:
    Discarded 0 reads containing invalid bases for a total 0 valid bases.       
    Discarded 206363 short reads for a total 748307569 bases.
    Discarded 0 reads containing repeat counts 256 or more for a total 0 bases. 
Read statistics for reads that will be used in this assembly:
    Total number of reads is 307254.
    Total number of raw bases is 24878481361.
    
I can upload more files if needed.
Thanks

[performance.log](https://github.com/user-attachments/files/17011056/performance.log)
Average read length is 80970.4 bases.
    N50 for read length is 128596 bases.
Found 0 reads with duplicate names.
Discarded from the assembly 0 reads with duplicate names.
Flagged 945 reads as palindromic out of 307254 total.
Palindromic fraction is 0.00307563
LowHash0 algorithm will use 2^30 = 1073741824 buckets.
Automatic settings for this iteration: minBucketSize 4, maxBucketSize 6
Alignment candidates after lowhash iteration 0: high frequency 608132, total 1425710, capacity 1425710.
Automatic settings for this iteration: minBucketSize 4, maxBucketSize 6
slurm-15143295.outShasta Release 0.12.0
2024-Sep-14 11:25:54.613492 Assembly begins.

**Comment by paoloshasta** [comment]
You should get a good assembly with these reads, provided you have sufficient coverage. Make sure to select the appropriate configuration depending on whether on not you ran Herro error correction on your reads.

I see from the output you attached that you only have 24 Gb of coverage, and if this is for a human genome it is probably insufficient coverage to generate a useful assembly. If it is a smaller genome you may be ok.

See [the slides](https://paoloshasta.github.io/shasta/Shasta-0.13.0.pdf) that accompany the Shasta 0.13.0 release for a summary of the results we obtained with these configurations on the reads released by ONT.

**Comment by paoloshasta** [comment]
I am not an expert in sequencing protocols, but ONT says they only used two flowcells to get 45x. The details of the protocol they used are in their announcement, and I am sure they would help if you are getting lower yield than they did. But your read N50 is excellent.

With your existing 25 Gb of coverage, you are at around 8X and that will not be sufficient to get anything close to a usable assembly. At best, it will be very fragmented and highly incomplete.

**Comment by colindaven** [comment]
In my hands shasta performs very reliably and well for plant genomes sequenced at 40-60X coverage (N50s 20 - 45 MB ) with one of the current configs (sometimes you have to test a few). 

I have had suboptimal fragmented assemblies with <0.5 MB N50 for large 4GB genomes with only 8-10X coverage.
8-10X coverage is not really sufficient for a good assembly, whatever the tool. I did have slightly better results with Flye though for very low coverage datasets. Canu might be another option (but it is slow).

You could also try correcting your reads with dorado correct, and then retrying shasta/flye/hifiasm.

**Comment by paoloshasta** [comment]
This is an attempt at assembling a human genome using 25 Gb of coverage, equivalent to about 8X coverage, or 4X per haplotype (this assembly configuration is for phased assembly). As I predicted in my post yesterday, this is insufficient coverage for a useful assembly. No changes of options and parameters, or error correction, will be able to make up for this. 

We tested the assembly configuration you are using at 45X and got a good assembly as described in [the slides](https://paoloshasta.github.io/shasta/Shasta-0.13.0.pdf) that accompany Shasta Release 0.13.0. I expect that you might still be able to get a usable assembly at around 25X to 30X, but it will be fragmented and not very complete. Your current cuverage is 3 to 4 times below that, so I think it is hopeless and your only recourse will be to do more sequencing.
